### PR TITLE
#85 Edit/Delete Agency Functionality

### PR DESCRIPTION
**Pull request by anigonzales**
Issue #85 

Issue Solution:
Added "name," "abbreviation" and "parent agency" input fields to allow the agency to be updated.
Added "admin delete agency" that only allows users to delete the agency. Confirmation shows up after clicking. If the user is not an admin, then it will be greyed out and show a tooltip on hover. 

Screenshots:
1.) The new fields and button for the edit agency modal
reg edit modal

2.) Input in one field will update the agency for that field only
change 1 input

3.) Agencies table reflects changes after form submission. (The column for the name is so long because I have all of the agencies on the table, so it adjusts the column size to the length of the longest name).
input changed table

4.) If the user is "admin," then the button will not be greyed out and will be clickable. After clicking it, a popup appears to confirm the deletion.
actual del popup

5.) If the user is not "admin," the button will grey out and not be clickable. A tooltip will appear on hover.
tooltip

Testing:
Added different inputs to each field and clicked submit, viewed those changes on the agencies table that the user can view.
The deletion button is disabled for any other user type. 
The tooltip only appears if the button is disabled. 
Deletion does not work if the agency has children.
Checked the usdr_grants db agency table in postgresql for the appropriate updates or deletions being reflected.

### CODE REVIEW

File: packages/client/src/components/Modals/EditAgency.vue [hunk 1]
```diff
@@ -44,6 +84,16 @@
...
           ></b-form-input>
         </b-form-group>
+        <form ref="form" @click="handleDelete">
+          <span id="disabled-wrapper" class="d-inline-block" tabindex="0">
+            <b-button v-bind:disabled="userRole !== 'admin'" style="pointer-events: none;" variant="danger">
+              Admin Delete Agency
+            </b-button>
+          </span>
+          <b-tooltip v-if="userRole !== 'admin'" target="disabled-wrapper" triggers="hover">
+            You cannot delete an agency with children. Reassign child agencies to continue deletion.
```
Comment (User1m) [inline packages/client/src/components/Modals/EditAgency.vue:94]: Tooltip msg is kinda confusing. 
Can you not delete an agency b/c you are not an admin? or b/c the agency has children?

File: packages/server/src/routes/agencies.js [hunk 2]
```diff
@@ -26,6 +27,39 @@ router.put('/:agency', requireAdminUser, async (req, res) => {
...
+    const result = await setAgencyAbbr(agency, abbreviation);
+    res.json(result);
+});
+
+router.put('/parent/:agency', requireAdminUser, async (req, res) => {
+    const { agency } = req.params;
+
+    // const { parentId } = req.body.parentId;
+    // console.log('reeeeeeqqqqq  ' + req.body.parentId);
+    const result = await setAgencyParent(agency, req.body.parentId);
```
Comment (User1m) [inline packages/server/src/routes/agencies.js:59]: Might want to cast `req.body.parentId` as a Number see [here](https://github.com/usdigitalresponse/usdr-gost/pull/116/files#diff-067094b50695ccd7d6188f26c341f3d1cd1213ffc6a762abcb069b5a7db841c4L38)

File: packages/client/src/components/Modals/EditAgency.vue (hunk 1 above)
Comment (anigonzales) [reply packages/client/src/components/Modals/EditAgency.vue:94]: If the user is not an admin or if the agency has children, the button will be disabled and the tooltip will appear on hover

**Review by User1m — APPROVED** [review]

**Comment by mhuang-usdr** [comment]
[QA]

A few things I noticed

**1. Upon clicking edit Agency - the fields are blank. They should be pre-populated with the existing values. **

So for this example, the name field should be prepopulated with "Administration: Administrative Services", the parent agency field should be pre-populated with "Administration, Department of", etc.

image


**2. I am still able to set the agency as its own parent - I shouldn't be able to do that.**

image

**3. A user should not be able to delete the agency they belong to, because that would orphan the user. Basically, since I am a member of the USDR agency, the "Delete Agency" button should be grayed out.** 

**4. Let's also rename "Admin Delete Agency" to simply "Delete Agency"**

image

**5. I am able to trigger the "Delete Agency" button even when I'm not clicking on the button itself. I am able to trigger it by clicking in the highlighted area as indicated below:**

image

**6. Is there any way to move the delete button to the bottom right corner of the modal?**
image

**7. The confirmation modal - let's rename the "OK" button to "Delete Agency" and make the delete button red.**

image

**8. For that same confirmation modal, I noticed that if click "Cancel" the entire "Edit Agency" modal disappears**. 
Let's change the behavior so that only the confirmation modal disappears when I click "Cancel" - the "Edit Agency" modal that is underneath should still remain. I recorded a quick Loom to show you what I mean.

https://www.loom.com/share/139ca4b065c24e98b1dc9bab04a1cb35


We're getting close - thank you!

**Pull request merged**